### PR TITLE
Modernize examples (ECMAScript and code style)

### LINKED ISSUE #17: Use ES2015 where appropriate
See Tom's link below for an animated GIF of what this issue is meant to accomplish.

@tmeasday commented on [Tue Feb 23 2016](https://github.com/meteor/meteor/issues/6305)

Note this isn't the modules stuff (see #6304), but instead ES2015 stuff.

Note we didn't do it for 1.2

---

@tmeasday commented on [Mon Feb 29 2016](https://github.com/meteor/meteor/issues/6305#issuecomment-189957787)

https://github.com/meteor/guide/blob/structure-content/content/code-style.md#javascript-style-guide

### PR DESCRIPTION
**Pull request by klaussner**
This pull request updates all code examples to use newer ECMAScript features and a more modern (and consistent) code style. If you think that I changed too much or if you prefer a different code style or formatting, I'd be happy to change it again, of course. :smile:

These are some of the changes I made:

* No more globals
* Arrow functions where `this` would be `window`/`global`. Publication functions, for example, still use `function`s, even if `this` isn't used.
* Shorter syntax for function definitions on object initializers
* More and consistent spacing to improve readability, for example using a space after opening and before closing braces for single line object literals.
* Single quotes instead of double quotes. The Guide and more recent code in the Docs also use single quotes.
* Template literals instead of string concatenation, except for multi-line strings where whitespace would be added at the beginning of each line.
* Camel case for publications, session variables, etc.
* Column size of 80 characters

Resolves #17

### CODE REVIEW

**Comment by hwillson** [comment]
Wow, this is awesome @klaussner - thanks for doing this!

File: source/api/methods.md [hunk 1]
```diff
@@ -139,19 +139,23 @@ will not throw an exception.  When the method is complete (which may or
...
 block until the method is complete.  It will eventually return the
 return value of the method, or it will throw an exception if the method
 threw an exception. (Possibly mapped to 500 Server Error if the
 exception happened remotely and it was not a `Meteor.Error` exception.)
 
-    // sync call
-    var result = Meteor.call('foo', 1, 2);
+```js
+// Synchronous call
+let result = Meteor.call('foo', 1, 2);
```
Comment (abernix) [inline source/api/methods.md]: Could probably be `const`? :)

Comment (klaussner) [inline source/api/methods.md]: Done. I also changed the `Assets.getText` example to use `const`. :)

**Comment by lorensr** [comment]
This is great, thanks! And I like single quotes 😉 

I liked how we had a `&nbsp;` in there 😊

**Pull request merged**